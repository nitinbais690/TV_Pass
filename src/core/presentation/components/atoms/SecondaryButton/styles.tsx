import { StyleSheet } from 'react-native';
import { appDimensionValues, appFontStyle } from 'core/styles/AppStyles';
import { scale } from 'qp-common-ui';

export const styles = (appColors: any) => {
    return StyleSheet.create({
        buttonStyle: {
            borderRadius: appDimensionValues.commonButtonHeight / 2,
            elevation: appDimensionValues.xxxxs,
            borderWidth: scale(1),
            borderStyle: 'solid',
            borderColor: appColors.secondaryButtonBorder,
            backgroundColor: appColors.secondaryButton,
            height: appDimensionValues.commonButtonHeight,
        },
        containerStyle: {
            flex: 1,
            height: appDimensionValues.commonButtonHeight,
        },
        titleStyle: {
            color: appColors.secondary,
            ...appFontStyle.buttonText,
        },
        unfocused: {
            backgroundColor: appColors.primaryVariant3,
            elevation: 0,
        },
    });
};
