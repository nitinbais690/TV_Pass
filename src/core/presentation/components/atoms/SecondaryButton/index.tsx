import React, { useState } from 'react';
import { Platform } from 'react-native';
import { Button as RNEButton, ButtonProps as RNEButtonProps } from 'react-native-elements';
import { styles } from './styles';
import { appFontStyle } from 'core/styles/AppStyles';
import useAppColors from 'core/presentation/hooks/use-app-colors';

interface ButtonProps extends RNEButtonProps {
    hasTVPreferredFocus?: boolean;
    secondaryTestID?: string;
    secondaryAccessibilityLabel?: string;
}

const SecondaryButton = (props: ButtonProps) => {
    const [focussed, setFocussed] = useState(false);
    let buttonStyles = styles(useAppColors());
    let fontStyles = appFontStyle;

    return (
        <RNEButton
            containerStyle={[buttonStyles.containerStyle, props.containerStyle]}
            buttonStyle={[
                buttonStyles.buttonStyle,
                Platform.isTV && !focussed ? buttonStyles.unfocused : {},
                props.buttonStyle,
            ]}
            titleStyle={props.titleStyle || [fontStyles.body3, buttonStyles.titleStyle]}
            {...props}
            onFocus={() => setFocussed(true)}
            onBlur={() => setFocussed(false)}
            hasTVPreferredFocus={props.hasTVPreferredFocus === false ? props.hasTVPreferredFocus : true}
            onPress={e => {
                if (props.onPress) {
                    // see here for more details: https://github.com/facebook/react-native/issues/25979
                    props.onPress(e);
                }
            }}
            testID={props.secondaryTestID}
            accessibilityLabel={props.secondaryAccessibilityLabel}
        />
    );
};

export default SecondaryButton;
