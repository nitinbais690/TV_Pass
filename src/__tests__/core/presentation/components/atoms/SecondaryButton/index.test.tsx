import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import SecondaryButton from 'core/presentation/components/atoms/SecondaryButton';

jest.mock('core/presentation/hooks/use-app-colors');

describe('secondaryButton', () => {
    test('renders correctly', () => {
        const { container } = render(<SecondaryButton />);
        expect(toJSON(container)).toMatchSnapshot();
    });
});
