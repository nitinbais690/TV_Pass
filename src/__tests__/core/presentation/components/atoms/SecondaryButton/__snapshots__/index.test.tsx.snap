// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`secondaryButton renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 3,
        "flex": 1,
        "height": 70,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      hasTVPreferredFocus={true}
      isTVSelectable={true}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#1A1D1E",
            "borderColor": "#35393F",
            "borderRadius": 35,
            "borderStyle": "solid",
            "borderWidth": 1.457142857142857,
            "elevation": 6,
            "flexDirection": "row",
            "height": 70,
            "justifyContent": "center",
            "padding": 8,
          }
        }
      />
    </View>
  </View>
</View>
`;
